### ISSUE TITLE
osc/ucx: destroying inuse region when using dynamic windows in OSU benchmark

### ISSUE THREAD

**Issue by devreal** [open-mpi/ompi#8774]
I'm running the OSU benchmark with Open MPI 4.0.5 in different configurations and found that UCX throws an error at the end:

```
$ mpirun -n 2 --mca osc ucx --mca btl ^openib ./build/mpi/one-sided/osu_get_latency -s flush -w dynamic
# OSU MPI_Get latency Test v5.6.2
# Window creation: MPI_Win_create_dynamic
# Synchronization: MPI_Win_flush
# Size          Latency (us)
1                      10.14
2                       9.04
4                       9.70
8                       9.61
16                      9.40
32                      9.45
64                      9.55
128                     9.79
256                     9.14
512                     9.26
1024                    9.32
2048                    9.48
4096                   10.32
8192                   10.54
16384                  10.88
32768                  11.66
65536                  13.46
131072                 16.25
262144                 21.91
524288                 33.39
1048576                55.73
2097152               100.71
4194304               190.05
[1617721591.848884] [r38c1t2n3:638363:0]         rcache.c:512  UCX  WARN  mlx5_1: destroying inuse region 0x1aff970 [0x18de000..0x18de010] g- rw ref 1 lkey 0x875e7b rkey 0x875e7b atomic_rkey 0x89900
[r38c1t2n3:638363:0:638363]      rcache.c:294  Assertion `region->refcount == 0' failed
==== backtrace (tid: 638363) ====
 0  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(ucs_handle_error+0x254) [0x14e5d46cd334]
 1  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(ucs_fatal_error_message+0xb8) [0x14e5d46ca2d8]
 2  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(ucs_fatal_error_format+0xde) [0x14e5d46ca3be]
 3  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(+0x288b8) [0x14e5d46cf8b8]
 4  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(+0x297ec) [0x14e5d46d07ec]
 5  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(ucs_class_call_cleanup_chain+0x76) [0x14e5d46d9b56]
 6  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucs.so.0(ucs_rcache_destroy+0x18) [0x14e5d46d1558]
 7  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/ucx/libuct_ib.so.0(uct_ib_md_close+0x5d) [0x14e5d45e225d]
 8  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/ucx-1.9.0/lib/libucp.so.0(ucp_cleanup+0x46) [0x14e5d4742456]
 9  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/openmpi-4.0.5/lib/openmpi/mca_osc_ucx.so(+0xad4d) [0x14e5d44d4d4d]
10  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/openmpi-4.0.5/lib/libmpi.so.40(ompi_osc_base_finalize+0x53) [0x14e5d7a9ccb3]
11  /zhome/academic/HLRS/hlrs/hpcjschu/opt-hawk/openmpi-4.0.5/lib/libmpi.so.40(ompi_mpi_finalize+0x5a1) [0x14e5d7a1d7b1]
12  ./build/mpi/one-sided/osu_get_latency() [0x401bc5]
13  /lib64/libc.so.6(__libc_start_main+0xf3) [0x14e5d6db77b3]
14  ./build/mpi/one-sided/osu_get_latency() [0x401cfe]
=================================
```

This is with UCX 1.9.0. The allocate and create window configurations work fine.

Note that this is different from #6987 where dynamic windows are broken on master/5.0.x whereas the error described here occurs on dynamic windows.

**Comment by brminich** [comment]
@janjust can you please take a look?

**Comment by janjust** [comment]
@brminich Will do

**Comment by hppritcha** [comment]
removing 4.0.x tag as fixed in the branch via #8922.  adding v4.1.x label as PR #8971 hasn't been merged yet.

**Comment by awlauria** [comment]
All releases branches merged. Closing.
